### ISSUE TITLE
Brush inputs/settings not calibrated for viewzoom and viewrotation (ascension, direction, dab-angle, speed, et al.)

### ISSUE THREAD

**Issue by briend** [mypaint/libmypaint#66]
Basically, a bunch of inputs and settings are not calibrated to how you are moving your hand/mouse or viewing the canvas (rotated or not).

This can create some odd scenarios as described in issue #65 where brushes start acting very strangely by just zooming in or out of the canvas.  I would like to propose that speed is somehow calculated to always reflect the "real world" speed of the mouse/stylus.  Maybe by factoring in the DPI of the monitor and the current zoom level.  I also think radius should be ignored; a large brush and small brush should have the same speed.

Also, when rotating the canvas the angles involved should be adjusted so that the brush continues to work as you'd expect if you'd rotated a physical canvas in front of you.

**Comment by briend** [comment]
Maybe changing the input range to 40 is a bad idea, breaks existing brushes.  If we just divided

**Comment by achadwick** [comment]
The brush engine is currently unaware of the zoom level.

**Comment by briend** [comment]
Not sure how I closed the issue.  Anyway, that makes sense.  So we'd have to pass zoom level just like pressure to brush lib?

**Comment by briend** [comment]
I'm working on a branch that passes zoom and view-rotation data as input data to libmypaint.  So now with the data the brush engine should be able to correct for them.  I've had success correcting Direction and Ascension, and will try to correct speed next:
https://github.com/briend/libmypaint/tree/view-zoom
https://github.com/briend/mypaint/tree/view-zoom
Any help/suggestions are appreciated!  I'm not a math-person so I basically trial-and-error 15 different formulas until the result is what I expect :-)

**Comment by briend** [comment]
Ok I have speed corrected for zoom in the branched in the previous post (yay!).  Funny the values stay between 0 and 4 so there is no need to change the brush settings.  I clamp at 4.0 just in case-- if you really whip the stylus I saw values around 5.0.

What I didn't realize is the gamma setting isn't supposed to let you have huge values-- it's supposed to CHANGE the values quickly (with higher gamma) and more slowly with lower gamma.   When I saw speeds of over 200 I assumed the gamma was responsible but really it was just quickly reporting the actual speed (yikes!).

Here's a demo video showing a brush w/ jitter controlled by speed-- works consistently at all zoom levels!
https://www.youtube.com/watch?v=JAQ5wUKbUOw&feature=youtu.be
